### ISSUE TITLE
Emscripten 1.39.19 compilation errors

### ISSUE THREAD

**Issue by kevzettler** [mamedev/mame#7031]
I am trying to compile on MacOS using:
`emmake make SUBTARGET=arcade TOOLS=1 REGENIE=1 -j5`

Emscripten version:
```
$ emcc --version
emcc (Emscripten gcc/clang-like replacement) 1.39.19 (f45bea21f3a8f74a68ed4e3e3d7e290807ee2aff)
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

There are some syntax and template errors
```
Compiling 3rdparty/lzma/C/Aes.c...
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:15:
In file included from ../../../../../src/lib/netlist/nl_config.h:11:
In file included from ../../../../../src/lib/netlist/plib/pexception.h:11:
In file included from ../../../../../src/lib/netlist/plib/pstring.h:11:
../../../../../src/lib/netlist/plib/ptypes.h:68:3: error: expected identifier
                EMSCRIPTEN
                ^
<built-in>:361:20: note: expanded from here
#define EMSCRIPTEN 1
                   ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:15:
In file included from ../../../../../src/lib/netlist/nl_config.h:11:
In file included from ../../../../../src/lib/netlist/plib/pexception.h:11:
In file included from ../../../../../src/lib/netlist/plib/pstring.h:11:
../../../../../src/lib/netlist/plib/ptypes.h:130:51: error: expected unqualified-id
                using os = std::integral_constant<ci_os, ci_os::EMSCRIPTEN>;
                                                                ^
<built-in>:361:20: note: expanded from here
#define EMSCRIPTEN 1
                   ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:17:
In file included from ../../../../../src/lib/netlist/plib/pmempool.h:11:
../../../../../src/lib/netlist/plib/palloc.h:626:60: error: expected ',' or '>' in template-parameter-list
        template <typename BASEARENA = aligned_arena, std::size_t PAGESIZE = 1024>
                                                                  ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:1:
In file included from ../../../../../src/emu/emu.h:23:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/vector:276:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/climits:41:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/limits.h:57:
/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc/limits.h:44:18: note: expanded from macro 'PAGESIZE'
#define PAGESIZE PAGE_SIZE
                 ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:1:
In file included from ../../../../../src/emu/emu.h:23:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/vector:276:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/climits:41:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/limits.h:57:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc/limits.h:8:
/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libc/musl/arch/emscripten/bits/limits.h:3:19: note: expanded from macro 'PAGE_SIZE'
#define PAGE_SIZE 16384
                  ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:17:
In file included from ../../../../../src/lib/netlist/plib/pmempool.h:11:
../../../../../src/lib/netlist/plib/palloc.h:626:60: error: template parameter missing a default argument
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:1:
In file included from ../../../../../src/emu/emu.h:23:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/vector:276:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/climits:41:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/limits.h:57:
/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc/limits.h:44:18: note: expanded from macro 'PAGESIZE'
#define PAGESIZE PAGE_SIZE
                 ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:1:
In file included from ../../../../../src/emu/emu.h:23:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/vector:276:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/climits:41:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx/limits.h:57:
In file included from /Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc/limits.h:8:
/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libc/musl/arch/emscripten/bits/limits.h:3:19: note: expanded from macro 'PAGE_SIZE'
#define PAGE_SIZE 16384
                  ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:17:
In file included from ../../../../../src/lib/netlist/plib/pmempool.h:11:
../../../../../src/lib/netlist/plib/palloc.h:626:33: note: previous default template argument defined here
        template <typename BASEARENA = aligned_arena, std::size_t PAGESIZE = 1024>
                                       ^
../../../../../src/lib/netlist/plib/palloc.h:660:75: error: too few template arguments for class template 'paged_arena'
        template <typename T, std::size_t ALIGN = PALIGN_VECTOROPT, typename A = paged_arena<>>//aligned_arena>
                                                                                 ^
../../../../../src/lib/netlist/plib/palloc.h:627:8: note: template is declared here
        class paged_arena : public arena_base<paged_arena<BASEARENA, PAGESIZE>, true, true>
              ^
../../../../../src/lib/netlist/plib/palloc.h:660:71: error: template parameter missing a default argument
        template <typename T, std::size_t ALIGN = PALIGN_VECTOROPT, typename A = paged_arena<>>//aligned_arena>
                                                                             ^
../../../../../src/lib/netlist/plib/palloc.h:660:44: note: previous default template argument defined here
        template <typename T, std::size_t ALIGN = PALIGN_VECTOROPT, typename A = paged_arena<>>//aligned_arena>
                                                  ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:15:
In file included from ../../../../../src/lib/netlist/nl_config.h:10:
../../../../../src/lib/netlist/plib/pconfig.h:70:33: note: expanded from macro 'PALIGN_VECTOROPT'
#define PALIGN_VECTOROPT        (32)
                                ^
Compiling 3rdparty/lzma/C/AesOpt.c...
Compiling src/mame/drivers/albazg.cpp...
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:18:
../../../../../src/lib/netlist/plib/ppmf.h:116:28: error: expected unqualified-id
                                        || ci::os() == ci_os::EMSCRIPTEN))                           ? 2 :
                                                              ^
<built-in>:361:20: note: expanded from here
#define EMSCRIPTEN 1
                   ^
In file included from ../../../../../src/mame/drivers/aleisttl.cpp:30:
In file included from ../../../../../src/devices/machine/netlist.h:16:
In file included from ../../../../../src/devices/machine/../../lib/netlist/nltypes.h:18:
../../../../../src/lib/netlist/plib/ppmf.h:116:13: error: no member named 'os' in 'plib::compile_info'
                                        || ci::os() == ci_os::EMSCRIPTEN))                           ? 2 :
                                           ~~~~^
../../../../../src/lib/netlist/plib/ppmf.h:110:12: error: no member named 'os' in 'plib::compile_info'
                                && ci::os() == ci_os::WINDOWS)                                   ? 0 :
                                   ~~~~^
../../../../../src/lib/netlist/plib/ppmf.h:129:2: error: static_assert failed due to requirement 'ppmf_internal::value == 2' "internal mismatch"
        static_assert(ppmf_internal::value == PHAS_PMF_INTERNAL, "internal mismatch");
        ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../src/lib/netlist/plib/ppmf.h:130:2: error: static_assert failed due to requirement 'integral_constant<int, 0>::value == 2' "type mismatch"
        static_assert(ppmf_type::value == PPMF_TYPE, "type mismatch");
        ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling 3rdparty/lzma/C/Alloc.c...
Compiling 3rdparty/lzma/C/Bcj2.c...
Compiling 3rdparty/lzma/C/Bra.c...
Compiling 3rdparty/lzma/C/Bra86.c...
Compiling 3rdparty/lzma/C/BraIA64.c...
Compiling 3rdparty/lzma/C/CpuArch.c...
Compiling 3rdparty/lzma/C/Delta.c...
Compiling src/mame/drivers/manohman.cpp...
11 errors generated.
em++: error: '/Users/kevzettler/code/emsdk/upstream/bin/clang++ -target wasm32-unknown-emscripten -D__EMSCRIPTEN_major__=1 -D__EMSCRIPTEN_minor__=39 -D__EMSCRIPTEN_tiny__=19 -D_LIBCPP_ABI_VERSION=2 -Dunix -D__unix -D__unix__ -Werror=implicit-function-declaration -Xclang -nostdsysteminc -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libcxx -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libcxxabi/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libunwind/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/compat -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libc/musl/arch/emscripten -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/local/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/SSE -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include -DEMSCRIPTEN -MMD -MP -MP -DNDEBUG -DCRLF=2 -DLSB_FIRST -DFLAC__NO_DLL -DPUGIXML_HEADER_ONLY -DMAME_NOASM -DLUA_COMPAT_ALL -DLUA_COMPAT_5_1 -DLUA_COMPAT_5_2 -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/compat -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libcxxabi/include -I../../../../../src/osd -I../../../../../src/emu -I../../../../../src/devices -I../../../../../src/mame -I../../../../../src/lib -I../../../../../src/lib/util -I../../../../../src/lib/netlist -I../../../../../3rdparty -I../../../../generated/mame/layout -I../../../../../3rdparty/libflac/include -I../../../../../3rdparty/glm -I../../../../../3rdparty/libjpeg -I../../../../../3rdparty/rapidjson/include -pipe -Werror -O3 -fno-strict-aliasing -Wno-unknown-pragmas -Wall -Wcast-align -Wundef -Wformat-security -Wwrite-strings -Wno-sign-compare -Wno-conversion -Wno-error=deprecated-declarations -Wno-cast-align -Wno-tautological-compare -Wno-unused-value -Wno-constant-logical-operand -fdiagnostics-show-note-include-stack -Wno-unknown-warning-option -Wno-extern-c-compat -Wno-unknown-attributes -Wno-ignored-qualifiers -Wno-missing-braces -Wno-self-assign-field -Wno-format-security -Wno-inline-new-delete -Wno-absolute-value -std=gnu89 -Wno-implicit-function-declaration -std=c++14 -Woverloaded-virtual -include ../../../../asmjs/obj/Release/emu.h -c ../../../../../src/mame/drivers/aleisttl.cpp -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include/SDL2 -I/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include/freetype2/freetype -c -o ../../../../asmjs/obj/Release/src/mame/drivers/aleisttl.o -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-cxx-exceptions -mllvm -emscripten-cxx-exceptions-allowed=_ZN15running_machine17start_all_devicesEv,_ZN12cli_frontend7executeEiPPc,_ZN8chd_file11open_commonEb,_ZN8chd_file13read_metadataEjjRNSt3__212basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE,_ZN8chd_file13read_metadataEjjRNSt3__26vectorIhNS0_9allocatorIhEEEE -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr' failed (1)
make[2]: *** [../../../../asmjs/obj/Release/src/mame/drivers/aleisttl.o] Error 1
make[1]: *** [alliedl] Error 2
make[1]: *** Waiting for unfinished jobs....
Compiling src/mame/drivers/ertictac.cpp...
Compiling 3rdparty/lzma/C/LzFind.c...
Compiling 3rdparty/lzma/C/Lzma2Dec.c...
Compiling 3rdparty/lzma/C/Lzma2Enc.c...
Compiling 3rdparty/lzma/C/Lzma86Dec.c...
Compiling 3rdparty/lzma/C/Lzma86Enc.c...
Compiling 3rdparty/lzma/C/LzmaDec.c...
Compiling 3rdparty/lzma/C/LzmaEnc.c...
Compiling src/mame/drivers/rmhaihai.cpp...
../../../../../3rdparty/lzma/C/LzmaEnc.c:1405:9: error: misleading indentation; statement is not part of the previous 'if' [-Werror,-Wmisleading-indentation]
        {
        ^
../../../../../3rdparty/lzma/C/LzmaEnc.c:1401:7: note: previous statement is here
      if (repIndex == 0)
      ^
1 error generated.
emcc: error: '/Users/kevzettler/code/emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -D__EMSCRIPTEN_major__=1 -D__EMSCRIPTEN_minor__=39 -D__EMSCRIPTEN_tiny__=19 -D_LIBCPP_ABI_VERSION=2 -Dunix -D__unix -D__unix__ -Werror=implicit-function-declaration -Xclang -nostdsysteminc -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/compat -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libc/musl/arch/emscripten -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/local/include -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/SSE -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include -DEMSCRIPTEN -fignore-exceptions -MMD -MP -MP -DNDEBUG -DCRLF=2 -DLSB_FIRST -DFLAC__NO_DLL -DPUGIXML_HEADER_ONLY -DMAME_NOASM -DLUA_COMPAT_ALL -DLUA_COMPAT_5_1 -DLUA_COMPAT_5_2 -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/compat -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/include/libc -I/Users/kevzettler/code/emsdk/upstream/emscripten/system/lib/libcxxabi/include -pipe -Werror -O3 -fno-strict-aliasing -Wno-unknown-pragmas -Wall -Wcast-align -Wundef -Wformat-security -Wwrite-strings -Wno-sign-compare -Wno-conversion -Wno-error=deprecated-declarations -Wno-cast-align -Wno-tautological-compare -Wno-unused-value -Wno-constant-logical-operand -fdiagnostics-show-note-include-stack -Wno-unknown-warning-option -Wno-extern-c-compat -Wno-unknown-attributes -Wno-ignored-qualifiers -Wno-missing-braces -Wno-self-assign-field -Wno-format-security -Wno-inline-new-delete -Wno-absolute-value -std=gnu89 -Wno-implicit-function-declaration -std=gnu99 -Wpointer-arith -Wstrict-prototypes -Wbad-function-cast -Wno-undef -Wno-strict-prototypes -c ../../../../../3rdparty/lzma/C/LzmaEnc.c -Xclang -isystem/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include/SDL2 -I/Users/kevzettler/code/emsdk/upstream/emscripten/cache/wasm/include/freetype2/freetype -c -o ../../../../asmjs/obj/Release/3rdparty/lzma/C/LzmaEnc.o -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr' failed (1)
make[2]: *** [../../../../asmjs/obj/Release/3rdparty/lzma/C/LzmaEnc.o] Error 1
make[1]: *** [7z] Error 2
Compiling src/mame/drivers/ssfindo.cpp...
Compiling src/mame/drivers/stellafr.cpp...
Compiling src/mame/machine/archimds.cpp...
Archiving libadp.a...
Archiving libalba.a...
Archiving libacorn.a...
make: *** [asmjs] Error 2
```

**Comment by DopefishJustin** [comment]
Confirmed, some of the recent netlist changes cause the build to fail. I can fix those issues easily enough but the resulting build crashes on load so I need to dig a bit further.

**Comment by couriersud** [comment]
I have left a number of comments on the commit.
In addition EMSCRIPTEN in pconfig.h and other headers need to be renamed.
Who on earth in the emscripten compiler team invented compiler defines not prefixed by underscores?

**Comment by DopefishJustin** [comment]
Should be resolved in d2b7643.
